release.sh

```diff
@@ -0,0 +1,144 @@
+
+# Perform a docker-machine release
+# GITHUB_TOKEN=XXXXX ./release.sh 0.X.X
+
+#GITHUB_USER=docker
+GITHUB_USER=jeanlaurent
+GITHUB_REPO=machine
+
+
+function display {
+    echo "🐳   $1"
+    echo ""
+}
+
+function checkError {
+    if [ "$?" -ne 0 ]; then
+        echo "😡   $1"
+        exit 1
+    fi
+}
+
+VERSION=$1
+GITHUB_VERSION="v$1"
+PROJECT_URL="git@github.com:$GITHUB_USER/$GITHUB_REPO"
+RELEASE_DIR=../release-$VERSION
+DOCKER_MACHINE=$GOPATH/src/github.com/docker/machine/bin/docker-machine
+
+if [ -z "$VERSION" ]; then
+  echo "Missing version argument"
+  exit 1
+fi
+
+if [[ -z ${GITHUB_TOKEN} ]]; then
+    echo "GITHUB_TOKEN missing"
+    exit 1
+fi
+
+if [[ ! -f ${DOCKER_MACHINE} ]]; then
+    echo "You must have a fresh build of docker-machine here : ${DOCKER_MACHINE}"
+    exit 1
+fi
+
+github-release --version > /dev/null 2>&1
+checkError "github-release is not installed, check https://github.com/aktau/github-release, aborting."
+
+display "Starting release $VERSION on $PROJECT_URL with token $GITHUB_TOKEN"
+
+ if [ -d "$RELEASE_DIR" ]; then
+   display "Cleaning up $RELEASE_DIR"
+   rm -rdf $RELEASE_DIR
+   checkError "Can't clean up."
+ fi
+
+ display "Cloning into $RELEASE_DIR from $PROJECT_URL"
+
+ mkdir -p $RELEASE_DIR
+ checkError "Can't create $RELEASE_DIR, aborting."
+ git clone -q $PROJECT_URL $RELEASE_DIR
+ checkError "Can't clone into $RELEASE_DIR, aborting."
+
+ cd $RELEASE_DIR
+
+ display "Bump version number to $VERSION"
+ sed -i.bak s/"$VERSION-dev"/"$VERSION"/g version/version.go
+ checkError "Sed borkage..., aborting."
+
+ git add version/version.go
+ git commit -q -m"Bump version to $VERSION" -s
+ checkError "Can't git commit the version upgrade, aborting."
+ rm version/version.go.bak
+
+ display "Checking machine 'release' exist"
+ $DOCKER_MACHINE ip release > /dev/null 2>&1
+ if [ "$?" -ne 0 ]; then
+   display "machine 'release' does not exist, creating it"
+   $DOCKER_MACHINE rm -f release 2> /dev/null
+   $DOCKER_MACHINE create -d virtualbox release
+ fi
+ eval $($DOCKER_MACHINE env release)
+ checkError "Machine 'release' is in a weird state, aborting."
+
+ display "Building in-container style"
+ USE_CONTAINER=1 make build-x
+ checkError "Build error, aborting."
+
+ display "Tagging and pushing tags"
+  git remote | grep prod
+  if [ "$?" -ne 0 ]; then
+    git remote add prod $PROJECT_URL
+  fi
+
+  git ls-remote --tags 2> /dev/null | grep ${GITHUB_VERSION}
+  if [ "$?" -ne 0 ]; then
+     git tag -d ${GITHUB_VERSION}
+     git push origin :refs/tags/${GITHUB_VERSION}
+  fi
+
+  display "Tagging release on github"
+  git tag ${GITHUB_VERSION}
+  git push prod ${GITHUB_VERSION}
+
+  display "Checking if release already exist"
+  github-release info \
+      --security-token  ${GITHUB_TOKEN} \
+      --user ${GITHUB_USER} \
+      --repo ${GITHUB_REPO} \
+      --tag ${GITHUB_VERSION}
+
+  if [ "$?" -ne 1 ]; then
+    display "Release already exist, trying to delete it."
+    github-release delete \
```

Comment:
I'm pretty wary of automatically deleting anything, just because it's very possible something unexpected might happen in the shell script and end up being a lot more destructive than anticipated.  Why not just count on removing things manually in this case?